data/firestore/src/main/java/io/github/droidkaigi/confsched2020/data/firestore/internal/FirestoreImpl.kt

```diff
@@ -76,9 +87,41 @@ internal class FirestoreImpl @Inject constructor() : Firestore {
             emit(counterRef)
         }
 
-        val thumbsUpSnapshot = setupThumbsUp.flatMapLatest {
-            it.toFlow()
-        }
+        var appliedCount = -1
+        val unappliedCountChannel = BroadcastChannel<Int>(10000)
+        val incrementFlow = thumbsUpEventChannel.asFlow()
+            .filter { it == sessionId }
+            .withIndex()
+            .onEach {
+                unappliedCountChannel.send(
+                    minOf(
+                        it.index - appliedCount,
+                        MAX_APPLY_COUNT
+                    )
+                )
+            }
+            // For firebase pricing
+            .debounce(INCREMENT_DEBOUNCE_MILLIS)
+            .map {
+                val result = minOf(it.index - appliedCount, MAX_APPLY_COUNT)
+                appliedCount = it.index
```

Comment:
`appliedCount` is incremented unlimited?

Reply:
Yes
appliedCount exists for it.index - appliedCount
it.index is incremented unlimited. So appliedCount is incremented unlimited too.

Reply:
Thank you, I understood.
`appliedCount` is a index of `thumbsUpEventChannel`that is already posted to firestore.
And `appliedCount`is a thumbs up count to should be incremented too.